test/integration/network_reserved_ips_test.go

```diff
@@ -2,150 +2,199 @@ package integration
+	}
+
+	// Verify the IP has been deleted
+	_, fetchDelErr := client.GetReservedIPAddress(context.Background(), resIP.Address)
+	if fetchDelErr == nil {
+		t.Errorf("Expected error when fetching deleted IP, got nil")
+	}
+
+	verifyDelList, verifyDelErr := client.ListReservedIPAddresses(context.Background(), NewListOptions(0, filter))
+	if verifyDelErr != nil {
+		t.Fatalf("Error listing IP addresses after deletion: %v", verifyDelErr)
+	}
+
+	if len(verifyDelList) != initialCount {
+		t.Errorf("Expected IP count to return to initial count, got %d, want %d", len(verifyDelList), initialCount)
 	}
 }
 
-// TestReservedIPAddresses_ListIPAddressesVariants tests various scenarios for listing IP addresses
+// TestReservedIPAddresses_ListIPAddressesVariants tests filters for listing IP addresses
 func TestReservedIPAddresses_ListIPAddressesVariants(t *testing.T) {
 	client, teardown := createTestClient(t, "fixtures/TestReservedIPAddresses_ListIPAddressesVariants")
 	defer teardown()
 
-	filter := ""
-	ipList, listIpErr := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
-	if listIpErr != nil {
-		t.Logf("Error listing ipaddresses, expected struct, got error %v", listIpErr)
-	} else {
-		if len(ipList) == 0 {
-			t.Logf("The customer has not reserved an IP %v", ipList)
+	// Reserve two IP addresses in us-east region
+	reservedIPs := make([]string, 2)
+	for i := 0; i < 2; i++ {
+		reserveIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{
+			Region: "us-east",
+		})
+		if err != nil {
+			t.Fatalf("Failed to reserve IP %d: %v", i+1, err)
+		}
+		reservedIPs[i] = reserveIP.Address
+		t.Logf("Successfully reserved IP %d: %s", i+1, reserveIP.Address)
+	}
+
+	// Defer cleanup of reserved IPs
+	defer func() {
+		for _, ip := range reservedIPs {
+			err := client.DeleteReservedIPAddress(context.Background(), ip)
+			if err != nil {
+				t.Errorf("Failed to delete reserved IP %s: %v", ip, err)
+			}
+		}
+	}()
+
+	// Create ListOptions with the filter for reserved IPs in us-east region
+	listOptions := linodego.ListOptions{
+		PageOptions: &linodego.PageOptions{
+			Page: 0,
+		},
+		Filter: "{\"reserved\":true,\"region\":\"us-east\"}",
+	}
+
+	ipList, err := client.ListIPAddresses(context.Background(), &listOptions)
+
+	if err != nil {
+		t.Fatalf("Error listing reserved IP addresses in us-east: %v", err)
+	}
+
+	t.Logf("Retrieved %d reserved IP addresses in us-east", len(ipList))
+
+	// Check if at least the two reserved IPs are in the list
+	foundReservedIPs := 0
+	for _, ip := range ipList {
+		if !ip.Reserved {
+			t.Errorf("Expected all IPs to be reserved, but found non-reserved IP: %s", ip.Address)
+		}
+		if ip.Region != "us-east" {
+			t.Errorf("Expected all IPs to be in us-east region, but found IP in %s region: %s", ip.Region, ip.Address)
+		}
+		for _, reservedIP := range reservedIPs {
+			if ip.Address == reservedIP {
+				foundReservedIPs++
```

Comment:
I can't find the check for `foundReservedIPs ` == `expected_ips`?

Reply:
I must have missed this check before committing. This check is now present.